### PR TITLE
Update to SDK 29

### PR DESCRIPTION
**Pull request by HashEngineering**
## Issue being fixed or feature implemented

NMA-597

Some lines of kotlin code had to be changed.

Travis config was changed also.  Let's see if the Travis CI tests pass.

This also will use dashj 0.17.5 (with protobuf-javalite)

## Related PR's and Dependencies

None

## Screenshots / Videos

None, no UI changes were made

## How Has This Been Tested?

- [x] QA (Mobile Team)

## Checklist:

- [x] I have performed a self-review of my own code and added comments where necessary
- [x] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: wallet/src/de/schildbach/wallet/ui/VerifySeedConfirmFragment.kt [hunk 1]
```diff
@@ -61,7 +61,7 @@ class VerifySeedConfirmFragment : VerifySeedBaseFragment() {
         view.findViewById<Toolbar>(R.id.toolbar).title = getString(R.string.verify)
 
         if (arguments?.containsKey("seed")!!) {
-            words.addAll(arguments!!.getStringArray("seed"))
+            words.addAll(arguments!!.getStringArray("seed")!!)
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/VerifySeedConfirmFragment.kt:64]: not sure why this became necessary to add `!!`.  Is there a better way?

perhaps getStringArray was changed to be able to return null.

File: wallet/src/de/schildbach/wallet/ui/widget/ReceiveInfoView.kt [hunk 2]
```diff
@@ -122,9 +122,9 @@ class ReceiveInfoView(context: Context, attrs: AttributeSet?) : ConstraintLayout
         try {
             val clipboardManager = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
             if (amount != null) {
-                clipboardManager.primaryClip = ClipData.newPlainText("Dash payment request", paymentRequestUri)
+                clipboardManager.setPrimaryClip(ClipData.newPlainText("Dash payment request", paymentRequestUri))
             } else {
-                clipboardManager.primaryClip = ClipData.newPlainText("Dash address", address.toBase58())
+                clipboardManager.setPrimaryClip(ClipData.newPlainText("Dash address", address.toBase58()))
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/widget/ReceiveInfoView.kt:127]: Did something change with setPrimaryClip such that it cannot be implicitly called with `primaryClip = x`

**Review by tomasz-ludek — COMMENTED** [review]
I added some comments but I don't insist on implementing them as they are out of the scope.

File: wallet/build.gradle [hunk 3]
```diff
@@ -37,7 +37,6 @@ dependencies {
     implementation 'androidx.preference:preference:1.1.0'
     implementation 'androidx.constraintlayout:constraintlayout:2.0.0-beta4'
     implementation "org.dashj:dashj-core:$dashjVersion"
-    implementation 'com.google.protobuf:protobuf-java:3.4.0'
```
Comment (tomasz-ludek) [inline wallet/build.gradle:40]: This was required to build the app with gradle 6.1.1

**Review by tomasz-ludek — APPROVED** [review]

File: wallet/src/de/schildbach/wallet/ui/widget/ReceiveInfoView.kt (hunk 2 above)
Comment (tomasz-ludek) [reply wallet/src/de/schildbach/wallet/ui/widget/ReceiveInfoView.kt:127]: They added the `@NonNull` annotation to the ClipData parameter in ClipboardManager.setPrimaryClip(ClipData) method, whereas getPrimaryClip() is annotated as `@Nullable` and it cause Kotlin to fail with the use of property access syntax.
Here is the full answer: https://stackoverflow.com/a/57128936/795721

**Review by tomasz-ludek — APPROVED** [review]

**Pull request merged**